### PR TITLE
optimises the performance of loading bani

### PR DESCRIPTION
**Pull request by inderpreetsingh**
* Added virtual scrolling to the controller
* appends slides when needed
* divides `printShabad` into separate modules
* cleans viewer.js a little bit

Now, bani loads much faster, and I see no performance issues. 

`known bugs`
* [x] if you drag the scrollbar on controller super fast to the end, it shows empty div. 
* [x] Possible edge case with bani length (what if next 10 lines are not in currently selected bani length)
* [x] sqlite does not work properly

_update_ : _solved_

`possible enhancements`
* Should I also remove the slides that are way up? Say, I am on verse 100th, should I remove top 30 verses? Or am I going into micro-optimizations?
* I can move the modes to separate constants file, felt like a weird pattern so still thinking on it. Any suggestions are welcome.

### CODE REVIEW

**Comment by inderpreetsingh** [comment]
A bug surfaced in this after adding sunder gutka bani length options, so I had to rewrite how bani length is handled. Now instead of hiding/showing verses from CSS, it reloads the whole bani with new bani length attribute.

File: www/js/search.js [hunk 1]
```diff
@@ -663,20 +663,21 @@ module.exports = {
...
-    const baniLengthList = ['short', 'medium', 'long', 'extralong'];
-    const baniLength = store.getUserPref('slide-layout.sunder-gutka.bani-length');
-    const line = rows.find((row) => row.baniLength[baniLengthList[baniLength]]);
-    return line ? line.ID : null;
-  },
-
   loadBani(BaniID) {
     const $shabadList = this.$shabad || document.getElementById('shabad');
+    const baniLength = store.get('userPrefs.toolbar.gurbani.bani-length');
+    const baniLengthCols = {
```
Comment (tsingh777) [inline www/js/search.js]: can you comment on this to provide some explanation

File: www/js/search.js [hunk 2]
```diff
@@ -726,76 +719,128 @@ module.exports = {
...
+
+    if (start) {
+      throughput = 10;
+    } else if (lineIndex > throughput) {
+      throughput = lineIndex + 1;
+    }
+
+    const end = start + throughput;
+
+    const lineHeight = 27;
```
Comment (tsingh777) [inline www/js/search.js]: Can numbers like these be defined as constants somewhere? having them inline could make refactoring a pain.

File: www/js/search.js [hunk 3]
```diff
@@ -726,76 +719,128 @@ module.exports = {
...
-    if (!apv) {
+    if (!apv && !start) {
       while (shabad.firstChild) {
         shabad.removeChild(shabad.firstChild);
       }
     }
 
-    let lineCount = 0;
-    rows.forEach((item) => {
+    shabad.parentNode.onscroll = (e) => {
```
Comment (tsingh777) [inline www/js/search.js]: please add some comments

File: www/js/search.js (hunk 2 above)
Comment (inderpreetsingh) [reply www/js/search.js]: Yes, I agree. I would also like to break down this file into multiple files. I think we will address all these issues in technical debt sprint that's coming up.

**Pull request merged**